### PR TITLE
Build binaries for environments when releasing a new version

### LINKED ISSUE #175: Build binaries for environments when releasing a new version
**What would you like to be added**:

Build binaries for environments when releasing a new version.

- yorkie-v0.1.4-darwin-amd64.zip
- yorkie-v0.1.4-linux-amd64.tar.gz
- yorkie-v0.1.4-linux-arm64.tar.gz
- yorkie-v0.1.4-linux-ppc64le.tar.gz
- yorkie-v0.1.4-windows-amd64.zip

etcd seems to build with [this script](https://github.com/etcd-io/etcd/blob/master/scripts/build-binary).

**Why is this needed**:
- Users can download and use Yorkie without building for their environment.

### PR DESCRIPTION
**Pull request by tahott**
**What this PR does / why we need it**:

Users can download and use Yorkie without building for their environment.

**Which issue(s) this PR fixes**:

Fixes #175 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: scripts/build-binary.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  exit 255
+fi
+
+set -u
+
+function main {
+  mkdir -p release
+  cd release
+
+  tarcmd=tar
```
Comment (dc7303) [inline scripts/build-binary.sh:18]: Is there any reason to manage it as a separate variable?

**Review by dc7303 — APPROVED** [review]
Thank you for your work!
If the things hackerwins suggested are fixed, it seems to be fine! :)

File: scripts/build-binary.sh (hunk 1 above)
Comment (tahott) [reply scripts/build-binary.sh:18]: No, there is no particular reason.
Should I remove the variable?

File: scripts/build-binary.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+set -e
+
+YORKIE_VERSION=$1
+
+if [ -z "$1" ]; then
+  echo "Usage: ${0} VERSION" >> /dev/stderr
+  exit 255
+fi
```
[the remark this answers is not in the document]
Comment (tahott) [reply scripts/build-binary.sh]: As per the review opinion, the code was modified and reflected.

File: Makefile [hunk 3]
```diff
@@ -44,4 +44,7 @@ docker:
 docker-latest:
 	docker build -t yorkieteam/yorkie:latest .
 
-.PHONY: tools proto build fmt lint test docker docker-latest
+release:
+	./scripts/build-binary.sh $(YORKIE_VERSION) '$(GO_LDFLAGS)'
```
Comment (hackerwins) [inline Makefile:48]:
```suggestion
	./scripts/build-binary.sh $(YORKIE_VERSION) $(GO_LDFLAGS)
```

**Review by hackerwins — APPROVED** [review]
LGTM. 
Thanks for the correction.

**Pull request merged**